server/src/main/java/org/opensearch/search/aggregations/bucket/terms/MultiTermsAggregator.java

```diff
@@ -414,6 +372,74 @@ public void close() {
         }
     }
 
+    static class MultiValuesSourceCollectorImpl implements MultiTermsValuesSourceCollector {
+
+        private final List<InternalValuesSourceCollector> collectors;
+        private final BytesStreamOutput scratch;
+        private final BytesKeyedBucketOrds bucketOrds;
+        private final BucketsAggregator aggregator;
+        private final LeafBucketCollector sub;
+
+        private final boolean collectViaAggregator;
```

Comment:
curious why we need this variable, have other optimization rely on this?

Reply:
Although, the C2 Compiler will optimise this eventually after multiple runs, I added this explicitly to avoid performing 2 null checks `aggregator != null && sub != null` for every composite key.